packages/core/core/generate-text/generate-text.ts

```diff
@@ -106,79 +110,165 @@ case of misconfigured tools.
 By default, it's set to 0, which will disable the feature.
      */
     maxToolRoundtrips?: number;
-  }): Promise<GenerateTextResult<TOOLS>> {
-  const retry = retryWithExponentialBackoff({ maxRetries });
-  const validatedPrompt = getValidatedPrompt({ system, prompt, messages });
 
-  const mode = {
-    type: 'regular' as const,
-    ...prepareToolsAndToolChoice({ tools, toolChoice }),
+    /**
+     * Optional telemetry information.
+     */
+    // This is meant to be both flexible for custom app requirements (metadata)
+    // and extensible for standardization (example: functionId, more to come).
+    telemetry?: {
+      functionId?: string;
+      metadata?: Record<string, AttributeValue>;
+    };
+  }): Promise<GenerateTextResult<TOOLS>> {
+  const baseTelemetryAttributes = {
+    'ai.model.provider': model.provider,
+    'ai.model.id': model.modelId,
+    // settings:
+    'ai.settings.maxRetries': maxRetries,
```

Comment:
Conventionally it should be `max_retries`. I know that this is annoying. Metadata is ok with camel-case since it's a user input. You can certainly break the custom here, this is just for context.

Reply:
I changed it and then reverted back. I think it makes sense to break the custom here for 2 reasons (unless it's a big concern):

1) easier code search since the rest of our code is in camelCase
2) smaller bundle size (we'd need a 'toSnakeCase' function otherwise)